datasets/conllpp/conllpp.py

```diff
@@ -0,0 +1,228 @@
+                            names=[
+                                "O",
+                                "B-PER",
+                                "I-PER",
+                                "B-ORG",
+                                "I-ORG",
+                                "B-LOC",
+                                "I-LOC",
+                                "B-MISC",
+                                "I-MISC",
+                            ]
+                        )
+                    ),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://github.com/ZihanWangKi/CrossWeigh",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        urls_to_download = {
+            "train": f"{_URL}{_TRAINING_FILE}",
+            "dev": f"{_URL}{_DEV_FILE}",
+            "test": f"{_URL}{_TEST_FILE}",
+        }
+        downloaded_files = dl_manager.download_and_extract(urls_to_download)
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_files["train"]}),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": downloaded_files["dev"]}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": downloaded_files["test"]}),
+        ]
+
+    def _generate_examples(self, filepath):
+        logging.info("⏳ Generating examples from = %s", filepath)
+        with open(filepath, encoding="utf-8") as f:
+            guid = 0
+            tokens = []
+            pos_tags = []
+            chunk_tags = []
+            ner_tags = []
+            for line in f:
+                if line.startswith("-DOCSTART-") or line == "" or line == "\n":
+                    if tokens:
+                        yield guid, {
+                            "id": str(guid),
+                            "tokens": tokens,
+                            "pos_tags": pos_tags,
+                            "chunk_tags": chunk_tags,
+                            "ner_tags": ner_tags,
+                        }
+                        guid += 1
+                        tokens = []
+                        pos_tags = []
+                        chunk_tags = []
+                        ner_tags = []
+                else:
+                    # conll2003 tokens are space separated
+                    splits = line.split(" ")
+                    assert len(splits) >= 4, line
+                    tokens.append(splits[0])
+                    pos_tags.append(splits[1])
+                    chunk_tags.append(splits[2])
+                    ner_tags.append(splits[3].rstrip())
+            # last example
+            yield guid, {
+                "id": str(guid),
+                "tokens": tokens,
+                "pos_tags": pos_tags,
+                "chunk_tags": chunk_tags,
+                "ner_tags": ner_tags,
+            }
```

Comment:
There is 14042 example in the train set of conllpp vs 14041 in the original conll2003 dataset.
This might come from this part which might yield an empty example.

Could you replace this part with this code please ?

```suggestion
            # last example
            if tokens:
                yield guid, {
                    "id": str(guid),
                    "tokens": tokens,
                    "pos_tags": pos_tags,
                    "chunk_tags": chunk_tags,
                    "ner_tags": ner_tags,
                }
```

And can you regenerate the dataset_infos.json file to check that we get 14041 examples this time ?
```
datasets-cli test ./datasets/conllpp --save_infos --ignore_verifications
```
`--ignore_verifications` skips the verification on the number of generated examples, so that it overwrites 14042 in the dataset_infos.json file to have 14041 instead.